src/services/identity/SitesService.ts

```diff
@@ -260,13 +322,18 @@ class SitesService {
     return updatedAt
   }
 
-  async getStagingUrl(sessionData: UserWithSiteSessionData): Promise<string> {
-    const { staging } = await this.getUrlsOfSite(sessionData, ["staging"])
-    if (!staging) {
-      throw new NotFoundError(
+  async getStagingUrl(
+    sessionData: UserWithSiteSessionData
+  ): Promise<string | NotFoundError> {
+    const siteUrls = await this.getUrlsOfSite(sessionData)
+    if (siteUrls instanceof NotFoundError) {
+      return new NotFoundError(
         `${sessionData.siteName} does not have a staging url`
       )
     }
+
```

Comment:
actually we could do a hierarchy of errors like so: `NotFoundError` -> `Site...` -> `StagingUrl...`